scrape_repo_information.py

```diff
@@ -0,0 +1,78 @@
+Stem counter by sushain
+"""
+
+CLIENT_ID = os.environ['GITHUB_CLIENT_ID']
+CLIENT_SECRET = os.environ['GITHUB_SECRET_CLIENT_ID']
+
+pairs = []
+Pair = collections.namedtuple('Pair', 'lg1 lg2 last_updated created direction repo stems')
+types = ["trunk", "nursery", "incubator", "staging"]
+params = urllib.parse.urlencode(dict({'client_id': CLIENT_ID, 'client_secret': CLIENT_SECRET}))
+
+def print_info(uri):
+    returned = get_info(uri)
+    return returned
+
+def get_info(uri):
+    try:
+        dictX = str((urllib.request.urlopen(uri)).read(), 'utf-8')
+        tree = xml.fromstring(dictX)
+    except:
+        return -1 # FIXME: error handling
+
+    return len(tree.findall("*[@id='main']/e//l"))
+
+for x in types:
+    html_url = "https://api.github.com/repos/apertium/apertium-%s/contents?" % x
+    html_data = urllib.request.urlopen(html_url + params)
+    html = json.loads(html_data.read())
+    for i in html:
+        if "apertium" in i["name"]:
+            try:
+                repo_name = x
+                direction = ""
+                lang_pair_name = i["name"]
+                #getting names
+                lg1 = lang_pair_name.split('-')[1]
+                lg2 = lang_pair_name.split('-')[2]
+                #getting into repository
+                link = "https://api.github.com/repos/apertium/%s/contents?" % lang_pair_name
+                repo_json = json.loads(urllib.request.urlopen(link + params).read())
+                for el in repo_json:
+                    if el["name"] == "modes.xml":
+                        download_url = el["download_url"]
+                        html_for_blob = urllib.request.urlopen(download_url).read()
+                        if lg1+'-'+lg2 in str(html_for_blob):
+                            direction += "<"
+                        if lg2+'-'+lg1 in str(html_for_blob):
+                            direction += ">"
+                    elif ".dix" in el["name"]:
+                        stems = print_info(el["download_url"])
+                    if direction == "><":
+                        direction = "<>"
+                commits_link = "https://api.github.com/repos/apertium/%s/commits?" % lang_pair_name
+                commits_html = urllib.request.urlopen(commits_link + params).read()
+                last_updated = json.loads(commits_html)[0]["commit"]["committer"]["date"]
+                created = json.loads(commits_html)[-1]["commit"]["committer"]["date"]
+                pair = Pair(created=created, last_updated=last_updated, lg1=lg1.strip(), lg2=lg2.strip(), direction=direction, repo=repo_name, stems=stems)
+                print(json.dumps(pair, default=lambda o: dict(o)))
+                pairs.append(pair)
+                time.sleep(0.1) #if you change this value you will get blocked likely
+
+            except (IndexError, TypeError):
+                pass #this happens if there is a weird entry in the repo
+
+if __name__ == "__main__":
```

Comment:
As I said in my comment, all the actually functionality should be in a main() function, not just these two lines. This is standard Python convention to avoid global pollution.